File: keybind-common/src/main/java/io/github/solusmods/eternalcore/keybind/api/KeybindingCategory.java
package io.github.solusmods.eternalcore.keybind.api;

import io.github.solusmods.eternalcore.keybind.ModuleConstants;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor(staticName = "of")
public class KeybindingCategory {
    private final String name;

    public String getCategoryString() {
        return String.format("%s.category.%s", ModuleConstants.MOD_ID, this.name);
    }
}

File: keybind-common/src/main/kotlin/io/github/solusmods/eternalcore/keybind/api/KeybindingCategory.kt
package io.github.solusmods.eternalcore.keybind.api

import io.github.solusmods.eternalcore.keybind.ModuleConstants

class KeybindingCategory private constructor(private val name: String) {

    val categoryString: String
        get() = String.format(
            "%s.category.%s",
            ModuleConstants.MOD_ID,
            this.name
        )

    companion object {
        @JvmStatic
        fun of(name: String): KeybindingCategory {
            return KeybindingCategory(name)
        }
    }
}
